### ISSUE TITLE
Add API documentation

### ISSUE THREAD

**Issue by cvigilv** [nvim-orgmode/orgmode#499]
### Does this feature exist in Emacs orgmode core?

N/A

### Feature value

Enable easier production of plugins for `orgmode.nvim`.

### Additional context

I have been trying my best to create some plugins for org-mode, but it has been a struggle due to (1) not seeing any reference to the API in the current documentation and (2) after finding it, I couldn't grasp my thoughts around it since its vague in description and short for the amount of functionality the plugin gives. 

Would gladly help in adding documentation and examples for this if needed since, as of today, the API could be more precise if you don't know the ins and outs of the plugin.

**Comment by jgollenz** [comment]
You are absolutely right. Documentation is one of our weak points at the moment. If you have the time to compile a basic wiki page for API docs, don't hesitate! Also don't worry about it being perfect, we can iteratively improve on it :+1:
